rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,260 @@
+             │    │           │           │       │                  │ │
+             │    └───────────┘           │       └──────────────────┘ │
+             │                            │                            │
+             │AWS Account: teleport-prod  │                            │
+             └────────────────────────────┼────────────────────────────┘
+                                          │
+                                          │                ┌───────────┐
+                                          │                │           │
+                                          └──────────────► │  Quay.io  │◄───────── quay.io/gravitational/teleport
+                                                           │           │
+                                                           └───────────┘
+```
+
+The infrastructure will live in the [cloud-terraform](https://github.com/gravitational/cloud-terraform) repository. The terraform for the `teleport-prod` account can be found [here](https://github.com/gravitational/cloud-terraform/tree/main/teleport-team/prod). Using AWS ECR and ECR Public allow us to rely on their managed infrastructure which reduces the operational complexity while enforcing our own security policies and allowing us to better audit changes to the environment. For more information on the pros and cons of alternatives, see [alternatives](#alternatives).
+
+Amazon ECR Public assigns a random, default registry alias when the first public repository is created. However, a custom alias, such as `teleport` or `gravitational` can be requested. This alias is the display name in the Amazon ECR Public Gallery. Since the recent [terraform registry rfd](https://github.com/gravitational/teleport-plugins/blob/e68f0b5c8f594575a2a8dfc1d54cb2f983a227ce/rfd/0002-custom-terraform-registry.md) chose to use a convention of `gravitational/teleport`. I propose we request the `gravitational` alias in order to stay consistent amongst our artifacts. Thus, the image would be available to pull from `public.ecr.aws/gravitational/teleport`. 
+
+### Security
+Most of the security standards that will be applied to the infrastructure for this RFD are defined in Cloud RFD 17 - [Artifact Storage Standards](https://github.com/gravitational/cloud/blob/9124947fdfb0773fa9bd567160481bed4ec84b7e/rfd/0017-artifact-storage-standards.md).
+
+All employee interaction with the registry and repositories will require our existing Okta SSO w/ MFA. Teleport employees will have read access to the internal ECR registry in order to test images before promoting them. Teleport employees will not have direct write access to images. Teleport Release Engineers responsible for the artifacts will have limited write access through an assumed role. This role must be logged to the audit logs. Release engineers shall have no ability to modify or change audit logs.
```

Comment:
This is good but there should be some manual escape hatch allowing us to publish images manually in case of an urgent fix or something which we sometimes need to do. If it's via "release engineer" credentials it's fine, let's just make sure that this process is documented somewhere.

Reply:
There is a `release-engineer-core-admin` role that is created that will have capabilities to do manual fixes in case of an urgent fix. I'll add more information on that to clarify.

Reply:
Added in https://github.com/gravitational/teleport/pull/13165/commits/2c4eb54ee73c19b03169fc596bc0a65ec0eb3111